plugin-core/src/main/scala/com/alpine/plugin/core/icon/OperatorIcon.scala

```diff
@@ -0,0 +1,33 @@
+package com.alpine.plugin.core.icon
+
+/**
+  * Created by Jennifer Thompson on 4/27/16.
+  */
+class OperatorIcon private (val filePrefix: String, val shapeName: String) {
+
+  def this(filePrefix: String, shape: IconShape) {
+    this(filePrefix, shape.getClass.getSimpleName.dropRight(1)) // dropRight(1) to remove the $ sign.
+  }
+
+}
+
+object OperatorIcon {
+  def apply(filePrefix: String, shape: IconShape) = new OperatorIcon(filePrefix, shape)
+}
+
+sealed trait IconShape
+
```

Comment:
Eventually this will need some comments. So people understand the workbench icon vs. icon prefix. Will be confusing.